src/datasets/packaged_modules/audiofolder/audiofolder.py

```diff
@@ -0,0 +1,161 @@
+import glob
+import os
+from typing import List
+
+import datasets
+from datasets.tasks import AutomaticSpeechRecognition
+
+
+class AudioFolderConfig(datasets.BuilderConfig):
+    """BuilderConfig for AudioFolder."""
+
+    def __init__(self, *args, sampling_rate=None, transcripts_filename="transcripts.txt", **kwargs):
+        if not sampling_rate:
+            raise ValueError(
+                "To load an audio dataset, you must provide the `sampling_rate` parameter. "
+                "For example: "
+                '>>> load_dataset("audiofolder", data_dir="path/to/dir", sampling_rate=16_000)'
+            )
+        self.sampling_rate = sampling_rate
+        self.transcripts_filename = transcripts_filename
+
+        super(AudioFolderConfig, self).__init__(*args, **kwargs)
```

Comment:
My motivation was that we don't want users to accidentally mess up with sampling rate but your argument that it's not a required param in the Audio feature (so that way it's not consistent) seems valid to me.